thrift/thrift_test.go

```diff
@@ -310,6 +315,137 @@ func TestThriftContextFn(t *testing.T) {
+		secondPostCalled := false
+
+		firstInterceptor := &mocks.Interceptor{}
+		secondInterceptor := &mocks.Interceptor{}
+
+		args.server.RegisterInterceptor(firstInterceptor)
+		args.server.RegisterInterceptor(secondInterceptor)
+
+		firstInterceptor.On(
+			"Pre", mock.Anything, serviceMethod, mock.Anything,
+		).Run(func(preArgs mock.Arguments) {
+			throwsArgs := preArgs.Get(2).(*gen.SimpleServiceThrowsArgs)
+			assert.Equal(t, arg, throwsArgs.Arg)
+			assert.False(t, secondPreCalled)
+			firstPreCalled = true
+		}).Return(nil)
+
+		secondInterceptor.On(
+			"Pre", mock.Anything, serviceMethod, mock.Anything,
+		).Run(func(preArgs mock.Arguments) {
+			assert.True(t, firstPreCalled)
+		}).Return(nil)
+
+		args.s1.On("Throws", mock.Anything, arg).Return(ret, nil)
+
+		firstInterceptor.On(
+			"Post", mock.Anything, serviceMethod, mock.Anything, mock.Anything, nil,
+		).Run(func(preArgs mock.Arguments) {
+			assert.True(t, firstPreCalled)
+			assert.True(t, secondPostCalled, "Second post was called first")
+			firstPostCalled = true
+			throwsResult := preArgs.Get(3).(*gen.SimpleServiceThrowsResult)
+			assert.Equal(t, ret, throwsResult.GetSuccess())
+		}).Return(nil)
+
+		secondInterceptor.On(
+			"Post", mock.Anything, serviceMethod, mock.Anything, mock.Anything, nil,
+		).Run(func(preArgs mock.Arguments) {
+			assert.False(t, firstPostCalled, "first post has not yet been called")
+			secondPostCalled = true
+		}).Return(nil)
+
+		_, err := args.c1.Throws(ctx, arg)
+
+		assert.NoError(t, err)
+		firstInterceptor.AssertExpectations(t)
+		secondInterceptor.AssertExpectations(t)
+	})
+}
+
+func TestPreInterceptorShortCircuiting(t *testing.T) {
+	withSetup(t, func(ctx Context, args testArgs) {
+		preError := fmt.Errorf("some error")
+
+		firstInterceptor := &mocks.Interceptor{}
+		errorInterceptor := &mocks.Interceptor{}
+		// This inteceptor should not get called so we call no .Ons on this mock
+		uncalledInterceptor := &mocks.Interceptor{}
+
+		firstInterceptor.On(
+			"Pre", mock.Anything, mock.Anything, mock.Anything,
+		).Return(nil)
+		errorInterceptor.On(
+			"Pre", mock.Anything, mock.Anything, mock.Anything,
+		).Return(preError)
+
+		firstTransformedError := fmt.Errorf("Tranformed")
+		secondTransformedError := &gen.SimpleErr{}
+
+		errorInterceptor.On(
+			"Post", mock.Anything, mock.Anything, mock.Anything, mock.Anything,
+			preError,
+		).Return(firstTransformedError)
+		firstInterceptor.On(
+			"Post", mock.Anything, mock.Anything, mock.Anything, mock.Anything,
+			firstTransformedError,
+		).Return(secondTransformedError)
+
+		args.server.RegisterInterceptor(firstInterceptor)
+		args.server.RegisterInterceptor(errorInterceptor)
+		args.server.RegisterInterceptor(uncalledInterceptor)
+
+		ctx, cancel := NewContext(time.Second)
+		defer cancel()
+
+		result, err := args.c1.Throws(ctx, "test")
+
+		assert.Equal(t, result, "")
+		assert.Equal(t, secondTransformedError, err)
+
+		firstInterceptor.AssertExpectations(t)
```

Comment:
nit: you could do
```go
mock.AssertExpectationsForObjects(t, firstInterceptor, errorInterceptor, uncalledInterceptor)
```
https://godoc.org/github.com/stretchr/testify/mock#AssertExpectationsForObjects